File: solved/042_sum_of_divisors.py
def main():
  # cdef int i, j
  N = int(input())

  P = [1 for i in range(N+1)]

  i = 2
  while i <= N:
    j = 1
    while i*j <= N:
      cnt = 0
      n = i*j
      while not n % i:
        n //= i
        cnt += 1 
      # print(i*j, cnt+1)
      P[i*j] *= (cnt+1)
      j += 1
    i += 1
    while i <= N and P[i] > 1:
      i += 1

  SUM = 0
  for i in range(1, N+1):
    SUM += i * P[i]
  print(SUM)

main()

File: solved/042_sum_of_divisors.cpp
#include <bits/stdc++.h>
#define POW(z) ((z)*(z))
#define MOD 998244353
using ll = long long;
using namespace std;

void print(auto &v) {
  for(auto itr=v.begin(); itr!=v.end(); itr++) cout << *itr << " ";
  cout << endl;
}

int main(void) {
  ll N;
  cin >> N;

  vector<ll> P(N+1, 1);
  
  ll i=2, j, cnt, n;
  while (i <= N) {
    j = 1;
    while (i*j <= N) {
      cnt = 0;
      n = i*j;
      while(!(n%i)) {
        n /= i;
        cnt++;
      }
      P[i*j] *= ++cnt;
      j++;
    }
    i++;
    while(i<=N && P[i]>1) i++;
  }
  ll sum = 0;
  for(ll i=1; i<=N; i++) {
    sum += i * P[i];
  }
  cout << sum << endl;
}


